[23기 심택준]ADD: 개인 페이지 작성 완료

taekjunsim opened the pull request:
## 수정 사항 간략한 한줄 요약
(여기에 수정 사항에 대한 간략한 한줄 요약/제목 작성해 주세요.)

7월 22일(목)
* 특정 태그 scss 추가
* Login 페이지 작성 완료
* Main 페이지 작성 완료
* Nav 컴포넌트 추가
		 
## 수정 사항들 자세한 내용
(여기에 수정 사항에 대한 자세한 내용을 작성해 주세요.)

* li 태그 = list-style: none 추가
* a 태그 = text-decoration: none / color: "black" 추가
* Login 페이지 scss 적용 확인
* Main 페이지 scss 적용 확인
* Nav 컴포넌트 적용 확인

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

taekjunsim committed:
"ADD : Login페이지, Main페이지, reset.scss 작성 완료"
- Login.js 페이지 작성 완료
- Login.scss 페이지 작성 완료
- Login.js 및 Login.scss 호환 확인

- Main.js 페이지 진행 중 (Nav component 미입력)
- Main.scss 페이지 진행 중 (Nav component 미입력)
- Main.js 및 Main.scss 호환 확인 (Nav 제외)

- reset.scss(Eric Meyer) + *, a태그 추가

taekjunsim committed:
"ADD : Nav 컴포넌트 추가, reset.scss 내용 삭제"
- Nav 컴포넌트 추가
- Nav.scss 추가
- Nav 호환 확인
- reset.scss 와 Nav충돌 발생해서 reset.scss 내용 삭제

taekjunsim committed:
"ADD = li태그 css수정, a태그 css수정"
- li태그: list-style: none,
- a태그 : text-decoration:none / color:black,

taekjunsim committed:
"ADD: Mission 1 - 사용자 입력 데이터 저장 기능 구현"
- 부모 컴포넌트: taekjunLogin.js
- 자식 컴포넌트: InputUserInfo.js
- 부모 컴포넌트에서 함수 생성 후 자식 컴포넌트에 전달

taekjunsim committed:
Add : Mission 2 - 로그인 버튼 활성화 기능 구현
- Login페이지에 validationCheck 함수 선언

taekjunsim committed:
Add : Mission 3 - 댓글 기능 구현

taekjunsim committed:
"ADD: input클릭 이벤트  추가"
- Main.scss: input focus outline 제거
(input에서 enter 클릭시 댓글 입력)
- inputBox: click이벤트 입력
- Main페이지: 클릭 시 엔터 감지 후 댓글 입력하는 함수 생성
- Main페이지: 버튼 클릭 후 input창 내용 삭제

nam2350 commented:
기능 구현을  빠르게 한 후 리팩토링을 진행해 코드의 변화되는 과정을 볼 수 있어서 좋았습니다.

ddodam commented:
택준님 수정파트를 보니 리팩토링실력이 👍👍 앞으로도 리팩토링 기대할게요😆 !!!!!

Kayoungkimjs commented:
전체적으로 코드가 깔끔하게 정리되어 있어 가독성이 좋고, 특히 클래스 네이밍 센스가 탁월하시네요! 클래스명만 보고도 어떤 속성인지 바로 알 수 있었어요! :)

siodev2929 commented:
택준님 댓글 pushComment 부분 concat 메서드 활용 참고할게요!!!

taekjunsim committed:
"Add : Mission 4 - 댓글 컴포넌트화, props로 데이터 전달 구현"

yeonuk-hwang commented on src/index.js:
```diff
@@ -3,6 +3,6 @@ import ReactDOM from 'react-dom';
 import Routes from './Routes';
 
 // import './styles/common.scss';
```
common.scss는 import 해야하지 않을까요?

taekjunsim replied:
확인! 수정완료했습니다 :)
체크박스도 수정했습니다!

yeonuk-hwang commented on src/pages/taekjun/Login/taekjunLogin.js:
```diff
@@ -0,0 +1,85 @@
+  getValue = e => {
+    const { name, value } = e.target;
+    if (name === 'id') {
+      value.indexOf('@') !== -1
+        ? this.setState({
+            idValue: value,
+          })
+        : this.setState({ idValue: value });
```
조건문이 true 든 false이든 모두 같은 동작을 수행하고 있네요!

taekjunsim replied:
확인! 개선했습니다 :)

yeonuk-hwang commented on src/pages/taekjun/Login/taekjunLogin.js:
```diff
@@ -0,0 +1,85 @@
+            pwValue: value,
+          })
+        : this.setState({ pwValue: value });
+    }
```
여기도 마찬가지입니다.

yeonuk-hwang commented on src/pages/taekjun/Login/taekjunLogin.js:
```diff
@@ -0,0 +1,85 @@
+  }
+
+  getValue = e => {
+    const { name, value } = e.target;
+    if (name === 'id') {
+      value.indexOf('@') !== -1
+        ? this.setState({
+            idValue: value,
+          })
+        : this.setState({ idValue: value });
+    }
+
+    if (name === 'pw') {
+      value.length > 4
+        ? this.setState({
+            pwValue: value,
+          })
+        : this.setState({ pwValue: value });
```
```suggestion
    const { name, value } = e.target;
    this.setState({[name]:value});
```
name 값을 id, pw가 아니라 state의 key값과 동일하게 idValue, pwValue로 설정하고 계산된 속성명 이용하면 위와 같이 표현 가능합니다.

taekjunsim replied:
이 부분은 이해가 잘 안 됩니다. 월요일에 여쭤보고 수정해도 될까요?

yeonuk-hwang commented on src/pages/taekjun/Login/taekjunLogin.js:
```diff
@@ -0,0 +1,85 @@
+  };
+
+  test = () => {
+    console.log(this.state.idValue, this.state.pwValue);
```
test 마친 콘솔은 삭제해주세요

taekjunsim replied:
확인! 수정완료했습니다 :)

yeonuk-hwang commented on src/pages/taekjun/Login/taekjunLogin.js:
```diff
@@ -0,0 +1,85 @@
+    }
+  };
+
+  test = () => {
```
함수명은 무슨 동작을 하는지 알아 볼 수 있게 지어주세요

yeonuk-hwang commented on src/pages/taekjun/Login/taekjunLogin.js:
```diff
@@ -0,0 +1,85 @@
+    return (
+      <div>
+        <div className="flexBox">
+          <div className="loginContainer">
```
h1~h6 요소를 감싸는데는 div보다 semantic 한 tag가 있어보입니다.

yeonuk-hwang commented on src/pages/taekjun/Login/taekjunLogin.js:
```diff
@@ -0,0 +1,85 @@
+                ></input>
+                <button
+                  className="loginButton"
+                  style={
```
inline-style이 아닌 className을 동적으로 바꾸면서 스타일링 해주세요

yeonuk-hwang commented on src/pages/taekjun/Login/taekjunLogin.js:
```diff
@@ -0,0 +1,85 @@
+                <button
+                  className="loginButton"
+                  style={
+                    this.state.idValidation && this.state.pwValidation
```
idValidation, pwValidation이란 state는 없어보입니다.

그리고 valid한지의 여부를 표시하는 값은 기존 state인 id, pw 값으로 계산이 가능해보이기에 state로 관리하지 않고 render시에 바로 계산해도 됩니다.

[공식문서 React로 사고하기](https://ko.reactjs.org/docs/thinking-in-react.html#step-3-identify-the-minimal-but-complete-representation-of-ui-state) 를 보시면 어떤 값들이 state가 되어야하는지에 대해 적혀있습니다.

> 각각 살펴보고 어떤 게 state가 되어야 하는 지 살펴봅시다. 이는 각 데이터에 대해 아래의 세 가지 질문을 통해 결정할 수 있습니다

> 1. 부모로부터 props를 통해 전달됩니까? 그러면 확실히 state가 아닙니다.

> 2. 시간이 지나도 변하지 않나요? 그러면 확실히 state가 아닙니다.

> 3. 컴포넌트 안의 다른 state나 props를 가지고 계산 가능한가요? 그렇다면 state가 아닙니다.

yeonuk-hwang commented on src/pages/taekjun/Login/taekjunLogin.scss:
```diff
@@ -0,0 +1,60 @@
+  justify-content: center;
+  width: 100%;
+  margin: 32px 0px 0px;
+  .loginContainer {
```
다른 css 선택자들끼리는 공백 주세요

yeonuk-hwang commented on src/pages/taekjun/Login/taekjunLogin.scss:
```diff
@@ -0,0 +1,60 @@
+        display: inline-flex;
+        flex-direction: column;
+        align-items: center;
+        input {
```
태그셀렉터는 지양해주세요, 만약 추후 다른  input tag가 생겼을 시 영향을 미칠 여지가 있기 때문에 클레스네임을 부여하셔서 스타일링을 하시는게 좋습니다.

yeonuk-hwang commented on src/pages/taekjun/Main/commentList.js:
```diff
@@ -0,0 +1,21 @@
+class CommentList extends Component {
+  render() {
+    return (
+      <li className="whoseComment" key={this.props.key}>
```
key props는 Arrary.map 을 통해서 엘리먼트를 생성할 때 각 엘리먼트를 구분짓기 위해서 사용하는 prop입니다.
지금 여기서는 key props을 전달할 이유가 없어보입니다.

yeonuk-hwang replied:
그리고 key props는 자식 컴포넌트한테 전달되지 않는 특수한 props이기에 this.props.key로 접근할 수 없습니다.

yeonuk-hwang commented on src/pages/taekjun/Main/commentList.js:
```diff
@@ -0,0 +1,21 @@
+        <img
+          className="commentHeartImg"
+          src="images/taekjun/heart.png"
+          alt="heartIcon"
```
alt 속성을 제일 먼저 기입해주세요

yeonuk-hwang commented on src/pages/taekjun/Main/taekjunMain.js:
```diff
@@ -0,0 +1,317 @@
+              </ul>
+
+              <article className="feedLayoutBox">
+                <header className="feedHeader">
```
header는 레이아웃 상 header를 표현하기 위한 tag가 아니라, h1~h6요소 등 제목요소를 감싸기 위한 태그입니다 적절치 않은 사용으로 보입니다.

yeonuk-hwang commented on src/pages/taekjun/Main/taekjunMain.js:
```diff
@@ -0,0 +1,317 @@
+                    {this.state.commentArray.map((content, index) => {
+                      return (
+                        <CommentList
+                          key={index}
```
key값을 index로 활용하는건 권장되지 않습니다.
변하지 않고 유일한 값으로 부여해주세요

yeonuk-hwang commented on src/pages/taekjun/Main/taekjunMain.js:
```diff
@@ -0,0 +1,317 @@
+                        value={this.state.newCommentValue}
+                        placeholder="댓글 달기..."
+                        onChange={this.newComment}
+                      ></input>
```
self-closing 가능해보입니다.

taekjunsim replied:
확인! 수정완료했습니다 :)

yeonuk-hwang commented on src/pages/taekjun/Main/taekjunMain.js:
```diff
@@ -0,0 +1,317 @@
+                          src="images/taekjun/user1.jpeg"
+                          alt="userProfileImage"
+                        />
+                        <div className="userInfo">
+                          <h6>areum</h6>
+                          <p className="whoIsFollowing">
+                            회원님을 팔로우합니다
+                          </p>
+                        </div>
+                      </div>
```
아래를 보면 계속 반복되는 부분으로 보입니다. 상수데이터 + Map 활용해서 표현해보세요

taekjunsim replied:
확인! 수정완료했습니다 :)

yeonuk-hwang commented on src/pages/taekjun/Main/taekjunMain.scss:
```diff
@@ -0,0 +1,311 @@
+/* -----main----- */
+li {
+  list-style: none;
+}
+
+a {
+  color: black;
+  text-decoration: none;
+}
```
전체적으로 영향을 미치게 될 속성인데 main.scss에 있으면 안됩니다. common.scss로 옮기는게 맞아보입니다.

taekjunsim replied:
확인! 수정완료했습니다 :)

yeonuk-hwang commented on src/pages/taekjun/Main/taekjunMain.scss:
```diff
@@ -0,0 +1,311 @@
+}
+
+.mainPage {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
```
mixin으로 만들어서 활용해보세요~

taekjunsim replied:
확인! 수정완료했습니다 :)

yeonuk-hwang commented on src/styles/reset.scss:
```diff
@@ -1,3 +1,8 @@
+/* http://meyerweb.com/eric/tools/css/reset/ 
+   v2.0 | 20110126
+   License: none (public domain)
+*/
+
```
추가할 필요 없어보입니다 이부분

taekjunsim replied:
확인! 수정완료했습니다 :)

yeonuk-hwang reviewed (commented):
수고하셨습니다 택준님, 리뷰 확인 및 반영해주세요
PR 본문에 체크박스 제대로 체크 안된 것 확인해주세요

taekjunsim committed:
"Fix: 리뷰 수정 사항 적용"
- reset.scss 불필요한 내용 삭제
- main.scss Mixin 적용
- main.scss 내용 일부 common.scss로 이동
- userList map함수 적용
- input 태그 셀프클로징 적용
- test 마친 콘솔 삭제 완료
- login페이지 내 같은 동작 수행하는 조건문 수정 완료
- common.scss import 완료
- 깃허브 체크박스 수정 완료
- idValue, pwValue 수정 사항 미반영
